lib/sles4sap_publiccloud.pm

```diff
@@ -1130,10 +1133,9 @@ sub wait_for_cluster {
     $args{max_retries} //= 7;
 
     while ($args{max_retries} > 0) {
-        my $hanasr_output = $self->run_cmd(cmd => 'SAPHanaSR-showAttr --format=script', quiet => 1);
         my $crm_output = $self->run_cmd(cmd => $crm_mon_cmd, quiet => 1);
 
-        my $hanasr_ready = check_hana_topology(input => calculate_hana_topology(input => $hanasr_output));
+        my $hanasr_ready = check_hana_topology(input => $self->get_hana_topology());
```

Comment:
rewrite it to directly use `get_hana_topology`. 

`hanasr_output` is no more needed for the `die` `record_info` that now is using `display_full_status`